### PR TITLE
rename __builtins.di to __importc_builtins.di

### PR DESCRIPTION
**Pull request by WalterBright**
Motivation: https://github.com/dlang/dmd/pull/16250#discussion_r1501905546

ping @JohanEngelen

### CODE REVIEW

File: compiler/src/dmd/id.d [hunk 1]
```diff
@@ -553,7 +553,7 @@ immutable Msgtable[] msgtable =
     { "_align", "align" },
     { "aligned" },
     { "__pragma", "pragma" },
-    { "builtins", "__builtins" },
+    { "builtins", "__importc_builtins" },
```
Comment (ibuclaw) [inline compiler/src/dmd/id.d:556]: From what I understand of the comment the problem is the other way around.

```
{ "importc_builtins", "__builtins" },
```

Comment (WalterBright) [reply compiler/src/dmd/id.d:556]: @JohanEngelen what do you say?

Comment (thewilsonator) [reply compiler/src/dmd/id.d:556]: given `{ "foo", "bar" }`, `foo` is the name used  in the DMD source code (`Id.foo`), `bar` is the name in D source code (`module bar;`)

Comment (JohanEngelen) [reply compiler/src/dmd/id.d:556]: I actually meant both:
- `builtins` as "module" name in DMD source code is very confusing for something that only relates to importC. I  like Iain's suggestion for a fix.
- the file name `__builtins.di` is unfortunate for importC, so I really like that this PR fixes that.

Comment (PetarKirov) [reply compiler/src/dmd/id.d:556]:
```suggestion
    { "importc_builtins", "__importc_builtins" },
```

Comment (WalterBright) [reply compiler/src/dmd/id.d]: ok

**Comment by JohanEngelen** [comment]
Nice.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
need to update frontend.h
```diff
diff --git a/home/circleci/dmd/compiler/src/dmd/frontend.h b/home/circleci/dmd/generated/linux/release/64/frontend.h
index 17500667dd..22e02c382f 100644
--- a/home/circleci/dmd/compiler/src/dmd/frontend.h
+++ b/home/circleci/dmd/generated/linux/release/64/frontend.h
@@ -8771,7 +8771,7 @@ struct Id final
     static Identifier* _deprecated;
     static Identifier* _align;
     static Identifier* aligned;
-    static Identifier* builtins;
+    static Identifier* importc_builtins;
```

**Pull request merged**